persistent/Database/Persist/Quasi.hs

```diff
@@ -332,7 +679,18 @@ mkEntityDef ps name entattribs lines =
     derives = concat $ mapMaybe takeDerives attribs
 
     cols :: [FieldDef]
-    cols = mapMaybe (takeColsEx ps) attribs
+    cols = reverse . fst . foldr k ([], []) $ reverse attribs
+    k x (!acc, !comments) =
+        case isComment =<< listToMaybe x of
+            Just comment ->
+                (acc, comment : comments)
+            Nothing ->
+                ( maybe id (:) (setFieldComments comments <$> takeColsEx ps x) acc
+                , []
+                )
```

Comment:
this is a similar thing but fortunately the code here is much simpler. We just collect comments until we hit a `FieldDef` and then we set the comments and move on.